src/forecast_common.jl

```diff
@@ -0,0 +1,65 @@
+"""
+Return the count of forecast windows.
+"""
+function get_count(forecast::Union{Deterministic, Probabilistic})
+    return length(get_data(forecast))
+end
+
+"""
+Return all initial times of the forecast.
+"""
+function get_initial_times(forecast::Union{Deterministic, Probabilistic})
+    return keys(get_data(forecast))
+end
+
+"""
+Return the initial_timestamp of the forecast.
+"""
+function get_initial_timestamp(forecast::Union{Deterministic, Probabilistic})
+    return first(keys(get_data(forecast)))
+end
+
+"""
+Return the forecast interval as a Dates.Period.
+"""
+function get_interval(forecast::Union{Deterministic, Probabilistic})
```

Comment:
Should we include here Scenarios, that forecasts need the same treatment as Probabilistic to make all the necessary constructors but I can make that after we merge this.

Reply:
I left out Scenarios because it is essentially not implemented.  I was thinking that whoever implements that functionality can make appropriate decisions.

Many of these methods require the data structures of Deterministic and Probabilistic and are guaranteed to fail for DeterministicSingleTimeSeries.